tests/registration_test.py

```diff
@@ -533,6 +533,114 @@ def test_strings_groupby_attach(self):
         self.assertIsInstance(sAttach.segments, ak.pdarray)
         self.assertIsInstance(sAttach.unique_keys, ak.Strings)
 
+    def test_string_property_accessor(self):
+        cleanup()
+        s = ak.array(["one", "two", "three", "123"])
+        s.register("sname")
+        ak.clear()
+        sAttach = ak.Strings.attach("sname")
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        s.get_bytes()
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+        s.get_offsets()
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
+
+        cleanup()
+        s = ak.array(["one", "two", "three", "123"])
+        s.register("sname")
+        ak.clear()
+        s.get_bytes()
+        s.get_offsets()
+        sAttach = ak.Strings.attach("sname")
+        self.assertListEqual(s.to_list(), sAttach.to_list())
+        self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
+        self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
```

Comment:
In my opinion, these 2 blocks are essentially testing the same thing, I think we can simplify it a bit to just one block. I made a few modifications to my suggestion. Adds validation that the property is no longer `None` after calling the get methods.

```suggestion
b = s.get_bytes()
self.assertTrue(s._bytes is not None)
o = s.get_offsets()
self.assertTrue(s._offsets is not None)
self.assertListEqual(s.to_list(), sAttach.to_list())
self.assertListEqual(s.get_bytes().to_list(), sAttach.get_bytes().to_list())
self.assertListEqual(s.get_offsets().to_list(), sAttach.get_offsets().to_list())
```